src/key.rs

```diff
@@ -168,9 +168,13 @@ impl str::FromStr for PublicKey {
     fn from_str(s: &str) -> Result<PublicKey, Error> {
         let mut res = [0u8; constants::UNCOMPRESSED_PUBLIC_KEY_SIZE];
         match from_hex(s, &mut res) {
-            Ok(constants::PUBLIC_KEY_SIZE) =>
-                PublicKey::from_slice(&res[0..constants::PUBLIC_KEY_SIZE]),
-            Ok(constants::UNCOMPRESSED_PUBLIC_KEY_SIZE) => PublicKey::from_slice(&res),
+            Ok(constants::PUBLIC_KEY_SIZE) => {
+                let bytes: [u8; constants::PUBLIC_KEY_SIZE] =
+                    res[0..constants::PUBLIC_KEY_SIZE].try_into().expect("length checked above");
```

Comment:
The `expect` message incorrectly mentions checking the length but that check has nothing to do with why `try_into` could theoretically fail. The reason it can't fail is that slicing uses the correct statically-known length.

(note that the check *also* doesn't prevent the slicing from panicking - that is prevented by the fact that `PUBLIC_KEY_SIZE <= UNCOMPRESSED_PUBLIC_KEY_SIZE`)

Reply:
Reworded to be more accurate